src/read.rs

```diff
@@ -48,10 +48,10 @@ mod tests {
     #[test]
     fn test() {
         let events = vec![
-            start_tag!("h1", attrs!["id" => "hello"]),
+            start_tag!("h1", attrs![ id: "hello", class: "fun" ]),
```

Comment:
One thing I do need is the ability to pass in an expression for the tag name, for converting from markdown I have `start_tag!(format!("h{}", level))` for creating the headers. If possible I think all the macros should use `expr` as their base case and have overloads that take in `ident` and simply `stringify!` it and recurse.